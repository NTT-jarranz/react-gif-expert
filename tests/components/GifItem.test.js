import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"; 

describe('GifItem test', () => {
    const title = "GifTitle";
    const url = "https://www.google.es/foto.jpg";

    test('should match snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}></GifItem>);
        expect(container).toMatchSnapshot();
    });

    test('should show image with URL and alternate text', () =>{
        render(<GifItem title={title} url={url}></GifItem>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    test('should show title in component', () => {
        render(<GifItem title={title} url={url}></GifItem>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})