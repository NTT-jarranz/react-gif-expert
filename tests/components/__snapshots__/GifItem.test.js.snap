// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GifTitle"
      src="https://www.google.es/foto.jpg"
    />
    <p>
      GifTitle
    </p>
  </div>
</div>
`;
